diff --git a/ci/build-win32.ps1 b/ci/build-win32.ps1
index c4cfc4edaf..a71624911a 100644
--- a/ci/build-win32.ps1
+++ b/ci/build-win32.ps1
@@ -121,6 +121,7 @@ opts.add_cmake_defines({
     'CMAKE_MSVC_RUNTIME_LIBRARY': 'MultiThreaded',
     'BUILD_SHARED_LIBS': 'OFF',
     'BUILD_TESTING': 'OFF',
+    'JPEGXL_VERSION': '0.12.0',
 })
 libjxl_proj = cmake.subproject('libjxl-cmake', options: opts)
 libjxl_dep = declare_dependency(dependencies: [
@@ -224,15 +225,18 @@ clone-recursive = true
     Set-Content -Path $project.Path -Value $content
 }
 
-meson setup build `
+$BUILD_DIR="build\windows\x86_64"
+meson setup $BUILD_DIR `
     --wrap-mode=forcefallback `
     -Ddefault_library=static `
+    --buildtype=release `
     -Dlibmpv=true `
-    -Dtests=true `
+    -Dtests=false `
+    -Dcplayer=false `
     -Dgpl=true `
     -Dffmpeg:gpl=enabled `
-    -Dffmpeg:tests=enabled `
-    -Dffmpeg:programs=enabled `
+    -Dffmpeg:tests=disabled `
+    -Dffmpeg:programs=disabled `
     -Dffmpeg:sdl2=disabled `
     -Dffmpeg:vulkan=auto `
     -Dffmpeg:libdav1d=enabled `
@@ -253,18 +257,14 @@ meson setup build `
     -Dlibplacebo:d3d11=enabled `
     -Dxxhash:inline-all=true `
     -Dxxhash:cli=false `
-    -Dluajit:amalgam=true `
     -Dd3d11=enabled `
     -Dvulkan=enabled `
-    -Djavascript=enabled `
+    -Djavascript=disabled `
     -Dwin32-smtc=enabled `
-    -Dlua=luajit `
+    -Dlua=disabled `
     -Ddrm=disabled `
     -Dlibarchive=disabled `
     -Drubberband=disabled `
     -Dwayland=disabled `
     -Dx11=disabled
-ninja -C build mpv.exe mpv.com libmpv.a
-cp ./build/subprojects/vulkan-loader/vulkan.dll ./build/vulkan-1.dll
-cp ./etc/mpv-*.bat ./build
-./build/mpv.com -v --no-config
+ninja.exe -C "$BUILD_DIR" install libmpv.a
