diff --git a/meson.build b/meson.build
index e2a2270..41a1ced 100644
--- a/meson.build
+++ b/meson.build
@@ -2317,8 +2317,12 @@ if conf.get('pthreads') == 0 and conf.get('w32threads') == 0 and conf.get('os2th
 endif
 
 if host_machine.system() == 'android' and get_option('jni').allowed()
-  if conf.get('jni_h') == 1 and not (conf.get('pthreads') == 1)
-    error('jni not found')
+  if conf.get('jni_h') == 1 and conf.get('pthreads') == 1
+    conf.set10('jni', true)
+  elif conf.get('jni_h') != 1
+    error('jni header not found')
+  else
+    error('jni requires pthreads')
   endif
 endif
 
@@ -2352,9 +2356,9 @@ conf.set10('fontconfig', conf.get('libfontconfig').is_odd())
 if conf.get('openssl') == 1
   test_openssl_dep = dependency('openssl', required: false, version: '>= 3.0.0', static: conf.get('static').is_odd())
   if test_openssl_dep.found()
-    is_blocked_by_license = not get_option('version3').disabled() or get_option('gpl').disabled() or not get_option('nonfree').disabled()
+    is_blocked_by_license = get_option('version3').disabled() or get_option('gpl').disabled() or get_option('nonfree').enabled()
     if is_blocked_by_license
-      error('OpenSSL >=3.0.0 requires -Dversion3=enabled')
+      error('OpenSSL >=3.0.0 requires -Dversion3=enabled and -Dgpl=enabled and -Dnonfree=disabled')
     endif
   else
     is_blocked_by_license = not get_option('gpl').disabled() and  get_option('nonfree').disabled()
