diff --git a/meson.build b/meson.build
index fb10be5078..42ece8ef04 100644
--- a/meson.build
+++ b/meson.build
@@ -1330,6 +1330,17 @@ if features['vulkan'] and features['x11']
      sources += files('video/out/vulkan/context_xlib.c')
 endif
 
+if host_machine.system() == 'darwin'
+    moltenvk = get_option('moltenvk').require(
+        features['vulkan'],
+        error_message: 'vulkan or moltenvk header could not be found!',
+    )
+    features += {'moltenvk': moltenvk.allowed()}
+    if features['vulkan'] and features['moltenvk']
+        sources += files('video/out/vulkan/context_moltenvk.m')
+    endif
+endif
+
 features += {'vk-khr-display': vulkan.type_name() == 'internal' or
                                cc.has_function('vkCreateDisplayPlaneSurfaceKHR', prefix: '#include <vulkan/vulkan_core.h>',
                                                dependencies: [vulkan])}
diff --git a/meson.options b/meson.options
index 52782f73d6..81bc9312ef 100644
--- a/meson.options
+++ b/meson.options
@@ -103,6 +103,7 @@ option('gl-dxinterop-d3d9', type: 'feature', value: 'auto', description: 'OpenGL
 option('ios-gl', type: 'feature', value: 'auto', description: 'iOS OpenGL ES interop support')
 option('videotoolbox-gl', type: 'feature', value: 'auto', description: 'Videotoolbox with OpenGL')
 option('videotoolbox-pl', type: 'feature', value: 'auto', description: 'Videotoolbox with libplacebo')
+option('moltenvk', type: 'feature', value: 'auto', description: 'Moltenvk context')
 
 # macOS features
 option('macos-10-15-4-features', type: 'feature', value: 'auto', description: 'macOS 10.15.4 SDK Features')
diff --git a/osdep/mac/meson.build b/osdep/mac/meson.build
index 1660528841..ba6aec1ccd 100644
--- a/osdep/mac/meson.build
+++ b/osdep/mac/meson.build
@@ -40,7 +40,8 @@ swift_compile = [swift_prog, swift_flags, '-module-name', 'swift',
                  '@INPUT@', '-I.', '-I' + source_root, '-I' + source_root / 'include',
                  '-I' + libplacebo.get_variable('includedir',
                             default_value: source_root / 'subprojects' / 'libplacebo' / 'src' / 'include'),
-                 '-I' + libavutil.get_variable('includedir')]
+                 '-I' + libavutil.get_variable('includedir',
+                            default_value: source_root / 'subprojects' / 'FFmpeg')]
 
 swift_targets = custom_target('swift_targets',
     input: swift_sources,
diff --git a/video/out/gpu/context.c b/video/out/gpu/context.c
index 60c0641a70..a1e3952785 100644
--- a/video/out/gpu/context.c
+++ b/video/out/gpu/context.c
@@ -50,6 +50,7 @@ extern const struct ra_ctx_fns ra_ctx_vulkan_xlib;
 extern const struct ra_ctx_fns ra_ctx_vulkan_android;
 extern const struct ra_ctx_fns ra_ctx_vulkan_display;
 extern const struct ra_ctx_fns ra_ctx_vulkan_mac;
+extern const struct ra_ctx_fns ra_ctx_vulkan_moltenvk;
 
 /* Direct3D 11 */
 extern const struct ra_ctx_fns ra_ctx_d3d11;
@@ -93,6 +94,9 @@ static const struct ra_ctx_fns *const contexts[] = {
 #if HAVE_X11
     &ra_ctx_vulkan_xlib,
 #endif
+#if HAVE_MOLTENVK
+     &ra_ctx_vulkan_moltenvk,
+#endif
 #if HAVE_COCOA && HAVE_SWIFT
     &ra_ctx_vulkan_mac,
 #endif
diff --git a/video/out/vulkan/context_mac.m b/video/out/vulkan/context_mac.m
index 281a4bdf2f..afd5439f37 100644
--- a/video/out/vulkan/context_mac.m
+++ b/video/out/vulkan/context_mac.m
@@ -75,7 +75,11 @@ static bool mac_vk_init(struct ra_ctx *ctx)
     if (!mpvk_init(vk, ctx, VK_EXT_METAL_SURFACE_EXTENSION_NAME))
         goto error;
 
-    p->vo_mac = [[MacCommon alloc] init:ctx->vo];
+    CAMetalLayer *layer = NULL;
+    if (ctx->vo->opts->WinID != -1)
+        layer = (CAMetalLayer *)(intptr_t)ctx->vo->opts->WinID;
+
+    p->vo_mac = [[MacCommon alloc] init:ctx->vo withLayer:layer];
     if (!p->vo_mac)
         goto error;
 
diff --git a/video/out/vulkan/context_moltenvk.m b/video/out/vulkan/context_moltenvk.m
new file mode 100644
index 0000000000..3b04c79c87
--- /dev/null
+++ b/video/out/vulkan/context_moltenvk.m
@@ -0,0 +1,95 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <CoreGraphics/CoreGraphics.h>
+#include <QuartzCore/CAMetalLayer.h>
+
+#include "common.h"
+#include "context.h"
+#include "utils.h"
+
+struct priv {
+    struct mpvk_ctx vk;
+    CAMetalLayer *layer;
+};
+
+static void moltenvk_uninit(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv;
+    ra_vk_ctx_uninit(ctx);
+    mpvk_uninit(&p->vk);
+}
+
+static bool moltenvk_init(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv = talloc_zero(ctx, struct priv);
+    struct mpvk_ctx *vk = &p->vk;
+    int msgl = ctx->opts.probing ? MSGL_V : MSGL_ERR;
+
+    if (ctx->vo->opts->WinID == -1) {
+        MP_MSG(ctx, msgl, "WinID missing\n");
+        goto fail;
+    }
+
+    if (!mpvk_init(vk, ctx, VK_EXT_METAL_SURFACE_EXTENSION_NAME))
+        goto fail;
+
+    p->layer = (__bridge CAMetalLayer *)(intptr_t)ctx->vo->opts->WinID;
+    VkMetalSurfaceCreateInfoEXT info = {
+         .sType = VK_STRUCTURE_TYPE_METAL_SURFACE_CREATE_INFO_EXT,
+         .pLayer = p->layer,
+    };
+
+    struct ra_ctx_params params = {0};
+
+    VkInstance inst = vk->vkinst->instance;
+    VkResult res = vkCreateMetalSurfaceEXT(inst, &info, NULL, &vk->surface);
+    if (res != VK_SUCCESS) {
+        MP_MSG(ctx, msgl, "Failed creating MoltenVK surface\n");
+        goto fail;
+    }
+
+    if (!ra_vk_ctx_init(ctx, vk, params, VK_PRESENT_MODE_FIFO_KHR))
+        goto fail;
+
+    return true;
+fail:
+    moltenvk_uninit(ctx);
+    return false;
+}
+
+static bool moltenvk_reconfig(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv;
+    CGSize s = p->layer.drawableSize;
+    ra_vk_ctx_resize(ctx, s.width, s.height);
+    return true;
+}
+
+static int moltenvk_control(struct ra_ctx *ctx, int *events, int request, void *arg)
+{
+    return VO_NOTIMPL;
+}
+
+const struct ra_ctx_fns ra_ctx_vulkan_moltenvk = {
+    .type           = "vulkan",
+    .name           = "moltenvk",
+    .reconfig       = moltenvk_reconfig,
+    .control        = moltenvk_control,
+    .init           = moltenvk_init,
+    .uninit         = moltenvk_uninit,
+};
